File: android/src/main/kotlin/com/bitmovin/player/PlayerManager.kt
package com.bitmovin.player

import android.content.Context
import android.util.ArrayMap
import com.bitmovin.player.api.Player
import com.bitmovin.player.api.PlayerConfig
import io.flutter.Log

object PlayerManager {
    val players: ArrayMap<String, Player> = ArrayMap<String, Player>()
    fun create(
        id: String,
        context: Context,
        playerConfig: PlayerConfig?,
    ): Player {
        val target = playerConfig?.let { Player.create(context, it) } ?: Player.create(context)
        players[id] = target
        return target
    }

    fun destroy(id: String) {
        players[id]?.let {
            it.onStop()
            it.destroy()
            players.remove(id)
        }
    }
}

File: ios/Classes/PlayerManager.swift
import BitmovinPlayer

public class PlayerManager {
  static let shared = PlayerManager()

  var players = [String: Player]()

  private init() {}

	func createPlayer(id: String, config: PlayerConfig?) {
		let player = PlayerFactory.create(playerConfig: config ?? PlayerConfig())
		players[id] = player
  }
    
	func destroy(id: String) {
		if let player = players[id] {
			player.destroy()
    	players.removeValue(forKey: id)
		}
	}
}
